internal/store/endpoint.go

```diff
@@ -195,7 +200,7 @@ func endpointMetricFamilies(allowAnnotationsList, allowLabelsList []string) []ge
 		),
 		*generator.NewFamilyGeneratorWithStability(
 			"kube_endpoint_ports",
-			"Information about the Endpoint ports.",
+			"(Deprecated from 2.14.0) Information about the Endpoint ports.",
```

Comment:
We have a dedicated field for deprecations: https://github.com/kubernetes/kube-state-metrics/blob/f3b7593031ba0c60cb35a897b954bdf123f41b39/pkg/metric_generator/generator.go#L37

IIRC if we use it, it will update the help message automatically

Reply:
+ tests need an update, looks good to merge then.

Reply:
Need to update help string here https://github.com/kubernetes/kube-state-metrics/blob/f3b7593031ba0c60cb35a897b954bdf123f41b39/pkg/metric_generator/generator.go#L84-L88

Reply:
Thanks, good catch!